pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -169,8 +175,13 @@ func (mi *StorageInitializerInjector) InjectModelcar(pod *v1.Pod) error {
 
 	// Create the modelcar that is used as a sidecar in Pod and add it to the end
 	// of the containers (but only if not already have been added)
+	modelContainer := mi.createModelContainer(image, constants.DefaultModelLocalMountPath)
 	if getContainerWithName(pod, ModelcarContainerName) == nil {
-		modelContainer := mi.createModelContainer(image, constants.DefaultModelLocalMountPath)
```

Comment:
why move outside the if condition?

Reply:
> why move outside the if condition?

I moved the declaration of modelContainer outside the if statement to improve code readability and help developers better understand its state changes. However, I realize this may cause unnecessary resource usage when the if condition isn't met. If you think it's better to keep the declaration inside the if, I'm happy to adjust. Thank you for your review

Reply:
I think it adds a new unneeded step since it will check every container.